vcd/resource_vcd_vapp_static_routing.go

```diff
@@ -0,0 +1,211 @@
+						"next_hop_ip": &schema.Schema{
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "IP Address of Next Hop router/gateway.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceVappNetworkStaticRoutingCreate(d *schema.ResourceData, meta interface{}) error {
+	return resourceVappNetworkStaticRoutingUpdate(d, meta)
+}
+
+func resourceVappNetworkStaticRoutingUpdate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	vappId := d.Get("vapp_id").(string)
+	vapp, err := vdc.GetVAppById(vappId, false)
+	if err != nil {
+		return fmt.Errorf("error finding vApp. %s", err)
+	}
+	vcdClient.lockParentVappWithName(d, vapp.VApp.Name)
+	defer vcdClient.unLockParentVappWithName(d, vapp.VApp.Name)
+
+	networkId := d.Get("network_id").(string)
+	staticRouting, err := expandVappNetworkStaticRouting(d)
+	if err != nil {
+		return fmt.Errorf("error expanding static routes: %s", err)
+	}
+	vappNetwork, err := vapp.UpdateNetworkStaticRouting(networkId, staticRouting, d.Get("enabled").(bool))
+	if err != nil {
+		log.Printf("[INFO] Error setting static routing: %s", err)
+		return fmt.Errorf("error setting static routing: %s", err)
+	}
+
+	d.SetId(vappNetwork.ID)
+
+	return resourceVappNetworkStaticRoutingRead(d, meta)
+}
+
+func resourceVAppNetworkStaticRoutingDelete(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	vappId := d.Get("vapp_id").(string)
+	vapp, err := vdc.GetVAppById(vappId, false)
+	if err != nil {
+		return fmt.Errorf("error finding vApp. %s", err)
+	}
+
+	vcdClient.lockParentVappWithName(d, vapp.VApp.Name)
+	defer vcdClient.unLockParentVappWithName(d, vapp.VApp.Name)
+
+	err = vapp.RemoveAllNetworkStaticRoutes(d.Get("network_id").(string))
+	if err != nil {
+		log.Printf("[INFO] Error deleting static routes: %s", err)
+		return fmt.Errorf("error deleting static routes: %s", err)
+	}
+
+	return nil
+}
+
+func resourceVappNetworkStaticRoutingRead(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	vappId := d.Get("vapp_id").(string)
+	vapp, err := vdc.GetVAppById(vappId, false)
+	if err != nil {
+		return fmt.Errorf("error finding vApp. %s", err)
+	}
+
+	vappNetwork, err := vapp.GetVappNetworkById(d.Get("network_id").(string), false)
+	if err != nil {
+		return fmt.Errorf("error finding vApp network. %s", err)
+	}
+
+	var rules []map[string]interface{}
+	if vappNetwork.Configuration.Features == nil || vappNetwork.Configuration.Features.StaticRoutingService == nil {
+		log.Print("no Static routes found.")
```

Comment:
I think this error would get lost in log without explicit loglevel prefix ([INFO] , [TRACE], etc)

Reply:
added